File: packages/native_pdf_renderer/lib/src/web/resources/page_repository.dart
import 'package:native_pdf_renderer/src/web/document/page.dart';
import 'package:native_pdf_renderer/src/web/pdfjs.dart';
import 'package:native_pdf_renderer/src/web/resources/repository.dart';
import 'package:uuid/uuid.dart';

final Uuid uuid = Uuid();

class PageRepository extends Repository<Page> {
  /// Register document in repository
  Page register(String? documentId, PdfJsPage renderer) {
    final page = Page(
      id: uuid.v1(),
      documentId: documentId,
      page: renderer,
    );
    set(page.id, page);
    return page;
  }

  @override
  void close(String? id) {
    get(id)!.close();
    super.close(id);
  }
}

File: packages/native_pdf_renderer/android/src/main/kotlin/io/scer/native_pdf_renderer/resources/PageRepository.kt
package io.scer.native_pdf_renderer.resources

import android.annotation.TargetApi
import android.graphics.pdf.PdfRenderer
import android.os.Build
import io.scer.native_pdf_renderer.document.Page
import io.scer.native_pdf_renderer.utils.randomID

@TargetApi(Build.VERSION_CODES.LOLLIPOP)
class PageRepository : Repository<Page>() {
    /**
     * Register page in repository
     * @returns page id
     */
    fun register(documentId: String, pageRenderer: PdfRenderer.Page): Page {
        val id = randomID
        val page = Page(id, documentId, pageRenderer)
        set(id, page)
        return page
    }

    public override fun close(id: String) {
        get(id).close()
        super.close(id)
    }
}